### PR TITLE
chore: bazel codeowners/approvers are configured

### LINKED ISSUE #11480: New Codeowners for bazel migration
# New Codeowners group for bazel migration

There are efforts to use Bazel as new build system for Magma (#11283).

## Problem

Changing the build system is a cross cutting concern requiring changes to (nearly) all components of Magma. However, there are no functional changes to the component, but only to the orthogonal subject of component builds. This separation is also clearly represented file-wise.

Being a cross cutting topic, there are often reviews requested, from _a lot_ of teams, which are sometimes (understandably) only reluctantly provided, as this is a new build system and the expertise is with the people involved in the migration in the beginning. This is a serious slow down for the migration.

## Solution

**We propose to temporarily create a new approvers group in order to speed up and increase efficiency of the Bazel migration process.**

In detail:
- The approvers group should be named `approvers-bazel`, i.e. the fully qualified name becoming `@magma/approvers-bazel`.
- We propose the following members to start with:
  - @electronjoe 
  - @themarwhal 
  - @nstng 
  - @LKreutzer 
- This new group should (mainly) gain control over all bazel relevant files, which are detailed in #11479. Roughly that are:
  - Basically all `*.bazel`, `*.bzl` files, which will affect files across multiple directories. 
  - Respective Bazel (only) directories, e.g. `/bazel/`, respective caches.
  - Bazel focused CI, e.g. Bazel build images, Bazel Github Action.

Once the Bazelification effort has stabilized, we plan to move back the ownership of the BUILD files to the wider team. Either giving it back to the subdirectory owner (the current approach) or giving it to `repo-magma-maintain` (all codeowners).

## Non-goals

- Changes to feature providing code or their codeowners.

### PR DESCRIPTION
**Pull request by nstng**
Signed-off-by: Nils Semmelrock <nils.semmelrock@tngtech.com>

chore: bazel codeowners/approvers are configured

## Summary

Proposal for CODEOWNER configuration for (temporary) bazel codeowners/approvers.

Precondition is that the group @magma/approvers-bazel exists.

The single lines are explained as follows:

**Essential**
* `/bazel/ @magma/approvers-bazel`
  * All custom bazel functionality
* `/.bazelrc @magma/approvers-bazel`
  * Bazel configuration
* `*.bazel @magma/approvers-bazel`
  * All BUILD.bazel files and WORKSPACE.bazel

**Semi-essential**
* `/.github/workflows/bazel-cache-push.yml @magma/approvers-bazel`
  * Bazel build and publish caches
* `/.github/workflows/bazel.yml @magma/approvers-bazel`
  * Base bazel build and test workflow
* `/.github/workflows/docker-builder-bazel-base.yml @magma/approvers-bazel`
  * Bazel base build workflow
* `/experimental/bazel-base/ @magma/approvers-bazel`
  * Bazel minimal external dependency environment
* `/.bazelignore @magma/approvers-bazel`
  * Bazel configuration

**Non-essential**
* `/.bazel-cache/ @magma/approvers-bazel`
  * basically only the .gitignore file in an empty folder - not essential (i.e. line can be removed if requested)
* `/.bazel-cache-repo/ @magma/approvers-bazel`
  * basically only the .gitignore file in an empty folder - not essential (i.e. line can be removed if requested)

**Notes**:
* All our `.bzl` files are in `/bazel`. This should be a design pattern, i.e. we do not need a general `*.bzl` entry.
* `agw-workflow.yml` contains some Bazel build steps, but is too much agw related.

## Test Plan

* Tests make sense after the group @magma/approvers-bazel is created.

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -61,6 +61,19 @@
 /docs/ @magma/approvers-docs
 /docs/readmes/proposals @electronjoe
 
+# approvers-bazel
+/.bazel-cache/ @magma/approvers-bazel
+/.bazel-cache-repo/ @magma/approvers-bazel7
```
Comment (sebathomas) [inline CODEOWNERS]: Typo
```suggestion
/.bazel-cache-repo/ @magma/approvers-bazel
```

Comment (nstng) [reply CODEOWNERS]: wups oO

**Review by LKreutzer — APPROVED** [review]
Done in pairing.

**Review by sebathomas — APPROVED** [review]

**Review by Neudrino — APPROVED** [review]

**Review by electronjoe — APPROVED** [review]

File: CODEOWNERS [hunk 2]
```diff
@@ -61,6 +61,18 @@
...
+/.bazel-cache/ @magma/approvers-bazel
+/.bazel-cache-repo/ @magma/approvers-bazel
+/.github/workflows/bazel-cache-push.yml @magma/approvers-bazel
+/.github/workflows/bazel.yml @magma/approvers-bazel
+/.github/workflows/docker-builder-bazel-base.yml @magma/approvers-bazel
+/bazel/ @magma/approvers-bazel
+/experimental/bazel-base/ @magma/approvers-bazel
+/.bazelignore @magma/approvers-bazel
+/.bazelrc @magma/approvers-bazel
+*.bazel @magma/approvers-bazel
```
Comment (themarwhal) [inline CODEOWNERS:74]: This will affect BUILD.bazel files across multiple directories. Once the Bazelification effort has stabilized, we plan to move back the ownership of the BUILD files to the wider team. Either giving it back to the subdirectory owner (the curent approach) or giving it to repo-magma-maintain (all codeowners) sound good to me. (I'm open to suggestions though, of couse.)
